Random NoneType error while downloading

Version 1.0.0 (I just installed it today)
Install via Python PIP
OS: Linux Mint 20.3
Python Version: 3.8.10
Command: mangadex-dl --login-username (redacted) --login-password (redacted) --save-as tachiyomi --cover original --unsafe https://mangadex.org/title/c1d990dd-9ff2-4e47-b93d-5e56e34c5069/digimon-xros-wars

I know tachiyomi is supposed to be default but I was seeing it using raw format so I enforced it.

I'm not sure what is triggering this.  If I restart the download it will go down and skip the files it grabbed then pickup just fine where it left off.

```
[ERROR] Unhandled exception, AttributeError: 'NoneType' object has no attribute 'startswith'
Traceback (most recent call last):
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/cli/__init__.py", line 35, in _main
    download(args)
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/cli/download.py", line 12, in download
    url(args, args.type)
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/cli/url.py", line 144, in __call__
    self.func(self.id, args)
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/cli/url.py", line 66, in download_manga
    dl_manga(*args)
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/main.py", line 331, in download
    fmt.main()
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/format/tachiyomi.py", line 61, in main
    success = downloader.download()
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/downloader.py", line 172, in download
    self._report(resp, length, round((t2 - t1) * 1000), False)
  File "/home/shinji/.local/lib/python3.8/site-packages/mangadex_downloader/downloader.py", line 254, in _report
    if resp.headers.get('x-cache').startswith('HIT'):
AttributeError: 'NoneType' object has no attribute 'startswith'

```

--- a/mangadex_downloader/downloader.py
+++ b/mangadex_downloader/downloader.py
@@ -249,17 +249,27 @@ class ChapterPageDownloader(FileDownloader):
 
     def _report(self, resp, size, _time, success):
         self.cleanup()
-        # Check if cached
-        cached = False
-        if resp.headers.get('x-cache').startswith('HIT'):
-            cached = True
-        
+
         # According to MangaDex devs
         # domain that not from mangadex.network are not allowed to report
         # so skip it
         if 'uploads.mangadex.org' in self.url:
             log.debug('Endpoint are not from mangadex.network, skipping report')
             return
+
+        # Check if cached
+        # If failed to retrieve images, mark cached as "False"
+        cached = False
+        # Fix #18
+        # Random NoneType error while downloading
+        # Whenever downloader get server error, Response headers are not parsed properly
+        if success:
+            cache_header = resp.headers.get('x-cache')
+
+            # Just in case something is happened
+            if cache_header is not None:
+                cached = cache_header.startswith('HIT')
+
 
         data = {
             'url': self.url,

